core/DataAccess/ArchiveSelector.php

```diff
@@ -61,80 +61,41 @@ public static function getArchiveIdAndVisits(ArchiveProcessor\Parameters $params
-        }
-
         $requestedPlugin = $params->getRequestedPlugin();
         $segment         = $params->getSegment();
         $plugins = array("VisitsSummary", $requestedPlugin);
 
         $doneFlags      = Rules::getDoneFlags($plugins, $segment);
         $doneFlagValues = Rules::getSelectableDoneFlagValues($includeInvalidated, $params);
 
-        $results = self::getModel()->getArchiveIdAndVisits($numericTable, $idSite, $period, $dateStartIso, $dateEndIso, $minDatetimeIsoArchiveProcessedUTC, $doneFlags, $doneFlagValues);
-
-        if (empty($results)) {
-            return false;
+        $results = self::getModel()->getArchiveIdAndVisits($numericTable, $idSite, $period, $dateStartIso, $dateEndIso, null, $doneFlags);
+        if (empty($results)) { // no archive found
+            return [false, false, false, false];
         }
 
-        $idArchive = self::getMostRecentIdArchiveFromResults($segment, $requestedPlugin, $results);
-
-        $idArchiveVisitsSummary = self::getMostRecentIdArchiveFromResults($segment, "VisitsSummary", $results);
-
-        list($visits, $visitsConverted) = self::getVisitsMetricsFromResults($idArchive, $idArchiveVisitsSummary, $results);
-
-        if (false === $visits && false === $idArchive) {
-            return false;
+        $result = self::findArchiveDataWithLatestTsArchived($results, $requestedPlugin, $segment);
+        if (!isset($result['idarchive'])
+            || !isset($result['nb_visits'])
+        ) {
+            return [false, false, false, true];
         }
 
-        return array($idArchive, $visits, $visitsConverted);
-    }
-
-    protected static function getVisitsMetricsFromResults($idArchive, $idArchiveVisitsSummary, $results)
-    {
-        $visits = $visitsConverted = false;
-        $archiveWithVisitsMetricsWasFound = ($idArchiveVisitsSummary !== false);
-
-        if ($archiveWithVisitsMetricsWasFound) {
-            $visits = $visitsConverted = 0;
+        if (!in_array($result['value'], $doneFlagValues)) { // the archive cannot be considered valid for this request (has wrong done flag value)
+            return [false, false, false, true];
         }
 
-        foreach ($results as $result) {
-            if (in_array($result['idarchive'], array($idArchive, $idArchiveVisitsSummary))) {
-                $value = (int)$result['value'];
-                if (empty($visits)
-                    && $result['name'] == self::NB_VISITS_RECORD_LOOKED_UP
-                ) {
-                    $visits = $value;
-                }
-                if (empty($visitsConverted)
-                    && $result['name'] == self::NB_VISITS_CONVERTED_RECORD_LOOKED_UP
-                ) {
-                    $visitsConverted = $value;
-                }
-            }
+        // the archive is too old
+        if ($minDatetimeArchiveProcessedUTC
+            && Date::factory($result['ts_archived'])->isEarlier(Date::factory($minDatetimeArchiveProcessedUTC))
```

Comment:
fyi getting some error here that `ts_archived` is null... and then throws an exception in Date::factory() that the date is not valid. Will try to have a quick look
![image](https://user-images.githubusercontent.com/273120/75500339-7f11ed00-5a31-11ea-99ec-195ac5a823a4.png)

Reply:
pushed a fix for it